### PR TITLE
installation: move non-dynamic setup to pyproject.toml

### PR DESCRIPTION
**Pull request by superlopuh**
Just to make the PR a little less scary, here's part 1/2 of the setup.py migration.

I tested that `pip install -e ".[extras]"` works with python 3.11

### CODE REVIEW

**Review by compor — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+"Issue Tracker" = "https://github.com/xdslproject/xdsl/issues"
+
+[project.scripts]
+xdsl-opt = "xdsl.tools.xdsl_opt:main"
+irdl-to-pyrdl = "xdsl.tools.irdl_to_pyrdl:main"
+xdsl-run = "xdsl.tools.xdsl_run:main"
+
+[tool.setuptools]
+platforms = ["Linux", "Mac OS-X", "Unix"]
+zip-safe = false
```
[the remark this answers is not in the document]
Comment (superlopuh) [reply pyproject.toml:23]: Excellent question, I'm just moveing settings around trying to preserve behaviour, but we might want to change that

[the remark this answers is not in the document]
Comment (AntonLydike) [reply pyproject.toml:23]: Ah, I see it was also part of the setup.py! Thanks!

**Review by math-fehr — APPROVED** [review]
Nice!

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (webmiche) [reply pyproject.toml:23]: If I had to guess, I'd say that this was just copied from somewhere once upon a time. Many of these setup thingis were :sweat_smile:

**Review by webmiche — APPROVED** [review]

**Pull request merged**